### ISSUE TITLE
Refactoring of the code base

### ISSUE THREAD

**Issue by danielnyan** [danielnyan/witch-fantasy#8]
Okay uhm... shit, how do I explain this? 
tl;dr: abandon all hope ye who lay thy eyes on this horrific codebase
rework codebase so that we can implement new spells to cast, e.g. allow players to cast Gun on their enemies. 

Currently, if you ever want to fire a missile, you will need to:  
1) Check under MovementController to see if the input is a given input, then obtain a prefab using a hardcoded variable in MovementController. 
2) After that, MovementController needs to dispatch to AnimatorController to handle the animations pertaining to the action of firing the missile. 
3) Oh yeah, MovementController also fires off a PunRPC command so that everyone in the server can see what's happening. Furthermore, it listens and sends variables in the stream so that everyone knows the current action of the player (for animation purposes)
4) But since firing missiles is a critical operation, MovementController doesn't rely on using PunRPC like a sane person. Instead, there is a specific event for it registered under GameEvents, so the MovementController calls a shitty API in GameEvents that dispatches the request, then GameEvents calls PhotonNetwork.RaiseEvent which raises the actual event that PlayerManager listens to. The PlayerManager examines where the firing player is according to the master client, instantiates a projectile with position and velocity, then tells everyone "Hey, I made this projectile here, ok thx bai". 
5) Oh yeah, and uhm... unless we're taking the game in a different direction, there's two colour variants for the projectile depending on whether you're in the purple or the green team. As such, PlayerManager also happens to shoulder the task of assigning the appropriate sprites for each weapon upon spawn.

To solve this, I'm thinking of separating the missile projectile into an external interface so that it can be treated as equipment. This is to be discussed. Ideally we should decouple the animation and multiplayer communication from MovementController and we should be able to add and remove this feature without breaking anything.

**Comment by danielnyan** [comment]
Updates: reducing scope of game to single player for now. Changing scope of game to include many spells that you can switch out, while putting flying controls on hiatus. May also have to revamp movement controls
